Day12/src/services/resultServices.ts

```diff
@@ -0,0 +1,184 @@
+    ) c ON studies_in.classid = c.classid
+    WHERE s.sid = $2
+   `;
+ return new Promise((resolve, reject) => {
+     sqlserver.query(query, [subjectId,studentId],(err: any, res: any) => {
+
+         if (err)
+             reject(err);
+         else
+             {
+                 if(res.rows.length==0)
+                   resolve(false);
+                  else  
+                  resolve(true);
+             }
+
+     })
+
+ })
+
+
+}
+
+async function addMarks(studentId :string , subjectId : string, marks : number) {
+
+    let query = `
+     INSERT INTO school.result(SUBJECTID,SID,marks) VALUES($1,$2,$3);
+    `;
+    return new Promise((resolve, reject) => {
+        sqlserver.query(query, [subjectId,studentId,marks], (err: any, res: any) => {
+
+            if (err)
+                reject(err);
+            else
+                resolve("marks added");
+
+        })
+
+    })
+    
+}
+
+async function updateMarks(studentId :string , subjectId : string, marks : number) {
+
+    let query = `
+      UPDATE school.result
+      SET marks = $1
+      WHERE SUBJECTID = $2 AND SID = $3
+    `;
+    return new Promise((resolve, reject) => {
+        sqlserver.query(query, [marks,subjectId,studentId], (err: any, res: any) => {
+
+            if (err)
+                reject(err);
+            else
+                resolve("marks added");
+
+        })
+
+    })
+    
+}
+
+async function getMarks(studentId :string ) {
+    
+   let query = `
+   SELECT s.subjectname , r.marks
+FROM school.result r
+LEFT JOIN school.subject s
+ON r.subjectid = s.subjectid
+WHERE r.sid = $1;
+   `
+   return new Promise((resolve, reject) => {
+    sqlserver.query(query, [studentId], (err: any, res: any) => {
+
+        if (err)
+            reject(err);
+        else {
+            
+            resolve(res.rows);
+
+        }
+
+    })
+
+})
+
+
+}
+
+async function getHighestMarksPerSubject(classId : string)
+{
+ 
+    let query = `
+    SELECT s.sid,s.name , a.subjectname , a.max_marks
+    FROM(
+    SELECT sb.subjectname,MAX(r.marks) AS max_marks
+    FROM school.student s
+    LEFT JOIN school.studies_in ON studies_in.sid = s.sid
+    LEFT JOIN school.class c ON studies_in.classid = c.classid
+    LEFT JOIN school.result r ON r.sid = s.sid
+    LEFT JOIN school.subject sb ON r.subjectid = sb.subjectid
+    WHERE c.classid = $1
+    GROUP BY subjectname
+    ) a
+    JOIN (
+	SELECT s.SID ,s.name , sb.subjectname , r.marks
+    FROM school.student s
+    LEFT JOIN school.result r ON r.sid = s.sid
+    LEFT JOIN school.subject sb ON r.subjectid = sb.subjectid
+   ) s
+
+    ON a.max_marks = s.marks AND a.subjectname = s.subjectname;
+
+   `;
+   return new Promise((resolve, reject) => {
+    sqlserver.query(query, [classId], (err: any, res: any) => {
+
+        if (err)
+            reject(err);
+        else {
+            
+            resolve(res.rows);
+
+        }
+
+    })
+
+})
+
+}
+
+async function getTop10Marks(classId : string) {
```

Comment:
we can make this generic, it can take count and return matching results. Now it only works return top 10